app.py

```diff
@@ -939,38 +982,81 @@ def snap_release(snap_name):
 @app.route('/account/snaps/<snap_name>/market/', methods=['POST'])
 @login_required
 def post_market_snap(snap_name):
-    whitelist = [
-        'title',
-        'summary',
-        'description',
-        'contact_url',
-        'keywords',
-        'license',
-        'price',
-        'blacklist_countries',
-        'whitelist_countries'
-    ]
-
     if 'submit_revert' in flask.request.form:
         flask.flash("All changes reverted.", 'information')
     else:
+        error_list = []
+        info = []
+        files = []
+        images_json = None
+
+        icon = flask.request.files.get('icon')
+        if icon is not None:
+            info.append(build_image_info(icon, 'icon'))
+            files.append(icon)
+
+        screenshots = flask.request.files.getlist('screenshots')
+        for screenshot in screenshots:
+            info.append(build_image_info(screenshot, 'screenshot'))
+            files.append(screenshot)
+
+        if not files:
+            # API requires a multipart request, but we have no files to push
+            # https://github.com/requests/requests/issues/1081
+            files = {'info': ('', dumps(info))}
+        else:
+            images_json = {'info': dumps(info)}
+
+        screenshots_list = snap_screenshots(
```

Comment:
Personally I'd call the variable simply `screenshots` - I think it's pretty implicitly obvious that if there are multiple screenshots it's going to be a list. It makes things like `for screenshot in screenshots:` a bit more poetic.